src/analyses/useAfterFree.ml

```diff
@@ -125,9 +142,18 @@ struct
     | StartOf lval
     | AddrOf lval -> warn_lval_might_contain_freed ~is_double_free transfer_fn_name ctx lval
 
-  let side_effect_mem_free ctx freed_heap_vars threadid =
-    let threadid = G.singleton threadid in
-    D.iter (fun var -> ctx.sideg var threadid) freed_heap_vars
+  let side_effect_mem_free ctx freed_heap_vars threadid joined_threads =
+    let side_effect_globals_to_heap_var heap_var =
+      let current_globals = ctx.global heap_var in
+      let joined_threads_to_add = match G.find_opt threadid current_globals with
+        | Some threads -> ConcDomain.ThreadSet.inter joined_threads threads
+        | None -> joined_threads
+      in
```

Comment:
Any reason this accumulation cannot happen inside the side-effect by default?

Reply:
In case a `join` occurs whenever one performs a side-effect via `ctx.sideg`, then I guess this accumulation here is indeed redundant. Furthermore, since the range of the map lattice, representing `G`, is a must set, I guess that a set intersection would be performed inside the side-effect by default.

If my assumptions are correct, then I guess we could it remove this.